lib/utils/payment_request.dart

```diff
@@ -10,16 +12,23 @@ class PaymentRequest {
     if (uri != null) {
       address = uri.path;
       amount = uri.queryParameters['tx_amount'] ?? uri.queryParameters['amount'] ?? "";
-      note = uri.queryParameters['tx_description']
-          ?? uri.queryParameters['message'] ?? "";
+      note = uri.queryParameters['tx_description'] ?? uri.queryParameters['message'] ?? "";
       scheme = uri.scheme;
     }
 
+    dynamic NanoUtil = nano!.getNanoUtil();
+
+    if (address.contains("nano")) {
+      amount = NanoUtil.getRawAsUsableStringInstance(amount, NanoUtil.rawPerNanoInstance) as String;
+    } else if (address.contains("ban")) {
+      amount = NanoUtil.getRawAsUsableStringInstance(amount, NanoUtil.rawPerBananoInstance) as String;
+    }
```

Comment:
this will throw an exception if nano is not enabled (ex. in Monero.com app)
also, same comment about using `getNanoUtil`